pisek/solution.py

```diff
@@ -20,6 +20,15 @@ def run_on_file(
         self.compile_if_needed()
         assert self.executable is not None
 
+        if (
+            output_file
+            and util.file_is_newer(output_file, self.executable)
+            and util.file_is_newer(output_file, input_file)
+        ):
+            # The output file is newer than both the executable and the input,
+            # so it should be up-to-date.
+            return program.RunResult.OK, output_file
```

Comment:
Přemýšlím, že když nějaké řešení spadne, tak po něm zbyde soubor a příště se nepustí a dostane WA (pokud nespadlo při ukončování 😀) místo RE. To mi přijde matoucí. Řešení je v takovém případě soubor mazat, což je ale k ničemu, pak to člověk nemůže debugovat. Jediné užitečné řešení mě napadá v takovém případě dát souboru hodně starou mtime – třeba 1.1.1970, ať to bije do očí. To není hezké, ale nevím, jak to udělat líp.

Reply:
To je pravda, ale je to tak okrajové, že myslím, že teď nestojí za to to řešit. Verdikt bude stejně, že to je špatně, a to by tak matoucí být nemělo. Přidal jsem ale do kódu poznámku o tom, že tohle se může stát.